packages/kit/test/utils.js

```diff
@@ -111,6 +111,9 @@ export const test = base.extend({
 	in_view: async ({ page }, use) => {
 		/** @param {string} selector */
 		async function in_view(selector) {
+			const visible = await page.isVisible(selector);
+			if (visible) return true;
+
```

Comment:
I remembered that `page.isVisible` checks if the element is anywhere on the page, regardless if it's in the viewport or not. For this function however, we want to check if the element is in the viewport or not. So perhaps we should remove these lines

Reply:
I actually think you should have two test, first that isVisible, and if it fails return false, if not then check if the element is actually in the current viewport,
just my opinion

Reply:
I don't think we necessarily need both checks and would probably opt for just one to keep the code simpler

Another thought though - just below this we call `page.$eval`. The docs for that have:

>This method does not wait for the element to pass actionability checks and therefore can lead to the flaky tests. Use [locator.evaluate(pageFunction[, arg, options])](https://playwright.dev/docs/api/class-locator#locator-evaluate), other [Locator](https://playwright.dev/docs/api/class-locator) helper methods or web-first assertions instead.

Perhaps we should call `page.waitForFunction` instead?

Reply:
Oof... sorry about that! I took another go at the logic with locator.